crow/src/lib.rs

```diff
@@ -215,6 +314,74 @@ mod tests {
         assert_eq!(expires_at, 1695314361);
         assert_eq!(topup_info.lnurlw, LNURL);
 
-        assert_eq!(topup_info.status, TopupStatus::READY)
+        assert_eq!(topup_info.status, TopupStatus::READY);
+
+        let topup = ListUncompletedTopupsTopup {
+            additional_info: Some("no_route".to_string()),
+            amount_sat: 42578,
+            amount_user_currency,
+            created_at: "2023-07-21T16:39:21.271+00:00".to_string(),
+            exchange_fee_rate: 0.014999999664723873,
+            exchange_fee_user_currency: 0.11999999731779099,
+            exchange_rate: 18507.0,
+            expires_at: "2023-09-21T16:39:21.919+00:00".to_string(),
+            id: "1707e09e-ebe1-4004-abd7-7a64604501b3".to_string(),
+            lightning_fee_user_currency: 0.0,
+            lnurl: LNURL.to_string(),
+            node_pub_key: "0233786a3f5c79d25508ed973e7a37506ddab49d41a07fcb3d341ab638000d69cf"
+                .to_string(),
+            status: topup_status_enum::FAILED,
+            user_currency: "eur".to_string(),
+        };
+
+        let topup_info = to_topup_info(topup).unwrap();
+
+        match topup_info.error {
+            None => {
+                panic!("Expected topup to have error code, was None")
+            }
+            Some(e) => match e {
+                TopupError::TemporaryFailure { code } => {
+                    assert_eq!(code, TemporaryFailureCode::NoRoute)
+                }
+                TopupError::PermanentFailure { .. } => {
+                    panic!("Expected topup to have temporary failure, was permanent")
+                }
+            },
+        }
```

Comment:
Simpler alternative:
```suggestion
        assert!(matches!(
            topup_info.error,
            Some(TopupError::TemporaryFailure {
                code: TemporaryFailureCode::NoRoute
            })
        ));
```